notebook/serverextensions.py

```diff
@@ -0,0 +1,197 @@
+        cfg = cm.get("jupyter_notebook_config")
+        server_extensions = (
+            cfg.setdefault("NotebookApp", {})
+            .setdefault("server_extensions", [])
+        )
+        if self._toggle_value:
+            if require not in server_extensions:
+                server_extensions.append(require)
+            else:
+                print("server extension already enabled")
+        elif self._toggle_value is None:
+            if require not in server_extensions:
+                print("server extension already disabled")
+            else:
+                server_extensions.remove(require)
+        cm.update("jupyter_notebook_config", cfg)
+
+    def toggle_server_extension_python(self, package):
+        m, server_exts = _get_server_extension_metadata(package)
+        for server_ext in server_exts:
+            require = server_ext['require']
+            self._toggle_server_extension(require)
+
+    def toggle_server_extension(self, require):
+        self._toggle_server_extension(require)
+
+    def start(self):
+
+        if not self.extra_args:
+            self.log.warn('Please specify a server extension/package to enable or disable')
+            sys.exit(1)
+        elif len(self.extra_args) > 1:
+            self.log.warn('Please specify one server extension/package at a time')
+            sys.exit(1)
+        if self.python:
+            self.toggle_server_extension_python(self.extra_args[0])
+        else:
+            self.toggle_server_extension(self.extra_args[0])
+
+
+class EnableServerExtensionApp(ToggleServerExtensionApp):
+
+    name = "jupyter serverextension enable"
+    description = "Enable a server extension using frontend configuration files."
+    _toggle_value = True
+
+
+class DisableServerExtensionApp(ToggleServerExtensionApp):
+
+    name = "jupyter serverextension disable"
+    description = "Disable an serverextension using frontend configuration files."
+    _toggle_value = None
+
+
+class ListServerExtensionsApp(BaseNBExtensionApp):
+
+    name = "jupyter serverextension list"
+    version = __version__
+    description = "List all server extensions known by the configuration system"
+
+    def list_server_extensions(self):
+        config_dirs = jupyter_config_path()
+        for config_dir in config_dirs:
+            self.log.info('config dir: {}'.format(config_dir))
+            cm = BaseJSONConfigManager(parent=self, config_dir=config_dir)
+            data = cm.get("jupyter_notebook_config")
+            server_extensions = (
+                data.setdefault("NotebookApp", {})
+                .setdefault("server_extensions", [])
+            )
+            if server_extensions:
+                self.log.info('    {}'.format(server_extensions))
+
+    def start(self):
+        self.list_server_extensions()
+
+
+_examples = """
+jupyter serverextension list                            # list all configured nbextensions
```

Comment:
Not sure... I like the parallelism between `jupyter nextensions` and `jupyter serverextension`, otherwise they have different "nested" levels...